sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -14302,6 +14302,211 @@ named_barrier_signal(uint8_t barrier_id, uint8_t producer_consumer_mode,
 
 /// @} sycl_esimd_memory_nbarrier
 
+/// @defgroup sycl_esimd_mask_compressed Mask compressed APIs.
+/// @ingroup sycl_esimd_memory
+
+/// @addtogroup sycl_esimd_mask_compressed
+/// @{
+
+/// template <typename Tx, int N,
+///       typename PropertyListT = oneapi::experimental::empty_properties_t>
+/// simd<Tx, N>
+/// mask_expand_load(Tx *p, simd_mask<N> mask, PropertyListT props = {});
+/// Mask expand load from USM memory location.
+/// The function reads data from a memory location using following algorithm:
+///
+/// \code{.cpp}
+///
+/// int Index = 0;
+/// for (int i = 0; i < N; ++i) {
+///       if (Mask[i]) {
+///           Result[i] = *(p + Index);
+///           ++Index;
+///       }
+/// }
+/// \endcode
+///
+///
+/// @tparam Tx is element type.
+/// @tparam N is the data size
+/// @param p is the base address for this operation.
+/// @param mask is the mask determining which elements will be read
+/// @param props The compile-time properties. Only cache hint
+/// properties are used.
```

Comment:
nit: mix between having a `.` at the end and not, i think we should have `.` to match other APIs